src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -399,6 +409,58 @@ bool CanApplyChange(DocumentId documentId)
             }
         }
 
+        private void NotifyRefactorChanges(IEnumerable<DocumentId> changedDocs, CodeAnalysis.Solution oldSolution)
+        {
+            // Without any services to notify there's no need to dig through the documents
+            if (!_refactorNotifyServices.Any())
+            {
+                return;
+            }
+
+            foreach (var changedDocId in changedDocs)
+            {
+                var newDocument = CurrentSolution.GetDocument(changedDocId);
+
+                // Documents without syntax tree won't have the annotations attached
+                if (!newDocument.SupportsSyntaxTree)
+                {
+                    continue;
+                }
+
+                var syntaxRoot = newDocument.GetSyntaxRootSynchronously(CancellationToken.None);
+                var symbolRenameNodes = syntaxRoot.GetAnnotatedNodes(RenameSymbolAnnotation.RenameSymbolKind);
+
+                foreach (var node in symbolRenameNodes)
+                {
+                    foreach (var annotation in node.GetAnnotations(RenameSymbolAnnotation.RenameSymbolKind))
+                    {
+                        var oldDocument = oldSolution.GetDocument(changedDocId);
+
+                        if (!(newDocument.TryGetSemanticModel(out var newSemanticModel) && newSemanticModel is object))
+                        {
+                            continue;
+                        }
+
+                        if (!(oldDocument.TryGetSemanticModel(out var currentSemanticModel) && currentSemanticModel is object))
+                        {
+                            continue;
+                        }
+
+                        var oldSymbol = RenameSymbolAnnotation.ResolveSymbol(annotation, currentSemanticModel.Compilation);
+                        var newSymbol = newSemanticModel.GetSymbolInfo(node).Symbol;
```

Comment:
Why isn't this calling GetDeclaredSymbol or something? We're annotating a _use_ and not the original declaration point?